crates/vendor/morphic_lib/src/analyze.rs

```diff
@@ -904,45 +905,59 @@ impl<'a> ForwardState<'a> {
                     .enumerate()
                     .map(|(slot_i, &heap_cell)| {
                         let mut val_summary = ForwardSccSlotSummary {
-                            pre_aliases: Set::new(),
-                            inductive_aliases: Set::new(),
-                            internal_aliases: Set::new(),
+                            pre_aliases: Set::default(),
+                            inductive_aliases: Set::default(),
+                            internal_aliases: Set::default(),
                             back_refs: Self::back_refs_in_states(
                                 &mut self.back_ref_states,
                                 block_version,
                                 heap_cell,
                             )
                             .clone(),
                         };
+                        let matching_slot: (ir::ValueId, u32) =
```

Comment:
The changes below here do not change the output of the code at all. This just changes how the output is generated. By using iterators here and some cloning (of said iterators), we are able to pre-allocate the correct size for each hash map. This is a large win for performance where rehashing this maps as they grew was taking significant time.